extension/src/popup/components/sendPayment/SendConfirm/TransactionDetails/hooks/useGetTxDetailsData.tsx

```diff
@@ -157,31 +157,52 @@ const getBuiltTx = async (
     networkDetails.networkPassphrase,
   );
   const sourceAccount: Account = await server.loadAccount(publicKey);
-  const operation = getOperation(
-    sourceAsset,
-    destAsset,
-    amount,
-    destinationAmount,
-    destination,
-    allowedSlippage,
-    path,
-    isPathPayment,
-    isSwap,
-    isFunded,
-    publicKey,
-  );
-  const transaction = new TransactionBuilder(sourceAccount, {
-    fee: xlmToStroop(fee).toFixed(),
-    networkPassphrase: networkDetails.networkPassphrase,
-  })
-    .addOperation(operation)
-    .setTimeout(transactionTimeout);
+  try {
+    const operation = getOperation(
```

Comment:
This change seems independent of the stellar-hd-wallet upgrade -- I'm fine with merging it in the same PR but I'm curious why a try/catch is necessary when constructing a transaction?